boa_engine/src/context/mod.rs

```diff
@@ -106,13 +104,13 @@ pub struct Context<'host> {
 
     /// ICU related utilities
     #[cfg(feature = "intl")]
-    icu: icu::Icu<'host>,
+    icu: icu::Icu,
 
-    host_hooks: MaybeShared<'host, dyn HostHooks>,
+    host_hooks: &'static dyn HostHooks,
```

Comment:
Why isn't this `Box` too, what if the user wants to store state?

Reply:
I think most uses of `HostHooks` override one or two methods without using any state at all. Using `&'static` avoids unnecessary allocations in that case, and users who want to have state inside those can just use `realm.host_defined()` or the (soon to be) `context.host_defined()`.

Reply:
Though, I'm not attached to this optimization per se, so if you have a different opinion, I can rollback to using a `Box<dyn HostHooks>`.

Reply:
That's completely true! Thank you for reminding me of that :)

Reply:
Oops, forgot that we need to use `Rc` because some methods take `context` by mutable reference, and `Rc<dyn Trait>` does allocate on zero sized types :(

Reply:
Every time we encounter this limitation again, I rethink if we should just take `Context` by immutable reference in the module loader and the host hooks...

Reply:
In that case, since we have `realm.host_defined()` and in the future `context.host_defined()`, I think it's fine :)